File: Java/300.java
import java.util.Scanner;

/**
 * 300 - Palabras Pentavocálicas - Iván - https://github.com/wildfireOfMine
 */
public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int prueba = sc.nextInt();
        sc.nextLine();
        for (int i = 1; i <= prueba; i++) {
            String frase = sc.nextLine();

            if (pentavolica(frase)) {
                System.out.println("SI");
            } else {
                System.out.println("NO");
            }
        }
    }

    public static boolean pentavolica(String frase) {
        boolean a = false;
        boolean e = false;
        boolean i = false;
        boolean o = false;
        boolean u = false;
        int j = 0;
        while (j < frase.length() && (!a || !e || !i || !o || !u)) {
            if (frase.charAt(j) == 'a') {
                a = true;
            } else if (frase.charAt(j) == 'e') {
                e = true;
            } else if (frase.charAt(j) == 'i') {
                i = true;
            } else if (frase.charAt(j) == 'o') {
                o = true;
            } else if (frase.charAt(j) == 'u') {
                u = true;
            }
            j++;
        }
        return (a) && (e) && (i) && (o) && (u);
    }
}

File: C++/300.cpp
#include <iostream>
#include <cctype>

using namespace std;

// 300 - Palabras Pentavólicas - Iván - https://github.com/wildfireOfMine

bool pentavocalic(string str) {
    bool a = false;
    bool e = false;
    bool i = false;
    bool o = false;
    bool u = false;
    for (int j = 0; j<str.length(); j++) {
        if (str[j] == 'a') {
            a = true;
        } else if (str[j] == 'e'){
            e = true;
        } else if (str[j] == 'i'){
            i = true;
        } else if (str[j] == 'o') {
            o = true;
        } else if (str[j] == 'u') {
            u = true;
        }
    }

    return a && e && i && o && u;
}

int main()
{
    int loop = 0;
    cin >> loop;
    for (int i = 0; i<loop; i++) {
        string phrase = "";
        cin >> phrase;
        if (pentavocalic(phrase)) {
            cout << "SI" << endl;
        } else {
            cout << "NO" << endl;
        }
    }
    return 0;
}
